src/lib/lessons.ts

```diff
@@ -4,23 +4,49 @@ import getAccessTokenFromCookie from '../utils/get-access-token-from-cookie'
 import {loadLessonComments} from './lesson-comments'
 import {sanityClient} from 'utils/sanity-client'
 import groq from 'groq'
+import some from 'lodash/some'
+import isEmpty from 'lodash/isEmpty'
 
+// next_up_url can be derived on the front-end from collection and
+// staff_notes_url seems to be null for all lessons
+// code_url is only used in a select few Kent C. Dodds lessons
 const lessonQuery = groq`
-*[_type == 'lesson' && slug == $slug][0]{
+*[_type == 'lesson' && slug.current == $slug][0]{
   title,
-  slug,
+  'slug': slug.current,
   description,
+  resource->_type == 'videoResource' => {
+    ...(resource-> {
+      'media_url': hslUrl,
+      'transcript': transcriptBody,
+      'transcript_url': transcriptUrl,
+      duration,
+      'subtitles_url': subtitlesUrl,
+    })
+  },
+  'free_forever': isCommunityResource,
+  'path': '/lessons/' + slug.current,
+  'thumb_url': thumbnailUrl,
+  'repo_url': repoUrl,
+  'code_url': codeUrl,
+  'createdAt': eggheadRailsCreatedAt,
+  'updatedAt': eggheadRailsUpdatedAt,
+  'publishedAt': eggheadRailsPublishedAt,
   'instructor': collaborators[0]-> {
-    'full_name': person->name,
-    'slug': person->slug.current,
-    'avatar_64_url': person->image.url,
-    'twitter_url': person->twitter
+    ...(person-> {
+      'full_name': name,
+      'slug': slug.current,
+      'avatar_64_url': image.url,
+      'twitter_url': twitter
+    }),
```

Comment:
This merge reference syntax is more performant than using an `->` for every reference we need off of person. The arrow in syntactic sugar for `*[_id == ^.thing._ref][0]`